base/db/R/clone_pft.R

```diff
@@ -0,0 +1,105 @@
+##' Duplicate existing pft with associated priors, species, and cultivars
+##'
+##' Creates a new pft that is a duplicate of an existing pft,
+##' including relationships with priors, species, and cultivars (if any) of the existing pft.
+##' This function mimics the 'clone pft' button in the PFTs record view page in the 
+##' BETYdb web interface for PFTs that aggregate >=1 species, but adds the ability to 
+##' clone the cultivar associations.
+##'
+##' @param parent.pft.name name of PFT to duplicate
+##' @param new.pft.name name for new PFT. Must not be the same as parent.pft.name
+##' @param new.pft.definition text for the new PFT's definition field.
+##' @param settings PEcAn settings list, used only for BETYdb connection parameters
+##' @return ID of the newly created pft in database, creates new PFT as a side effect
+##' @author David LeBauer, Chris Black
+##' @examples \dontrun{
+##' clone_pft(parent.pft.name    = "tempdecid",
+##'           new.pft.name       = "mytempdecid",
+##'           new.pft.definition = "mytempdecid is a new pft",
+##'           settings = pecan_settings_list)
+##' }
+clone_pft <- function(parent.pft.name,
+                      new.pft.name,
+                      new.pft.definition,
+                      settings){
```

Comment:
I still think that, in general, we should move away from `settings` arguments in favor of more explicit arguments, since doing it this way obscures the things actually required for this to work if you want to do it outside of the PEcAn workflow. But that's just, like, my opinion, man.